src/components/Common/Modal/ViewFollowModal/index.tsx

```diff
@@ -0,0 +1,69 @@
+import { useEffect, useState } from 'react';
+import { useRecoilValue } from 'recoil';
+
+import Avatar from '@/components/Common/Avatar';
+import { DEFAULT_PROFILE_IMAGE } from '@/constants/profile';
+import { userAtom } from '@/recoil/user';
+import { getUserDetailsById } from '@/services/Follow/follow';
+
+import {
+  FollowButton,
+  FollowButtonWrapper,
+  NoListMessage,
+  UserList,
+  UserListItem,
+  UserName,
+  ViewFollowWrapper,
+} from './style';
+
+export default function ViewFollowModal() {
+  const user = useRecoilValue(userAtom);
+  const [activeList, setActiveList] = useState('팔로잉');
+  const [userDetails, setUserDetails] = useState([]);
+
+  useEffect(() => {
+    async function fetchUserDetails(ids: string[]) {
+      const details = await Promise.all(
+        ids.map((id: string) => getUserDetailsById(id)),
+      );
+      setUserDetails(details);
+    }
```

Comment:
Promise.all로 모든 id에 대한 유저 정보를 가져오셨군요
아주 좋습니다 👍👍👍

여기서 getUserDetailsById의 return 값이 User 또는 null이라서
null인 데이터는 필터링으로 제거해주는게 필요해보여요!

```suggestion
    async function fetchUserDetails(ids: string[]) {
      const details = await Promise.all(
        ids.map((id: string) => getUserDetailsById(id)),
      );
      
      const filteredDetails = details.filter(
        (detail): detail is User => detail != null,
      );
      setUserDetails(filteredDetails);
    }
```

여기서 `detail is User`를 넣어준 이유는
TypeScript는 filter 메소드가 null 값을 제거한다는걸 몰라서
해당 부분을 넣어주어야 filteredDetails가 User[]으로 추론이 된다고 합니다!

저도 요 개념은 좀 생소해서 [관련된 블로그 자료](https://horang98.tistory.com/20) 공유드립니다~

Reply:
null 필터링이 필요하군요 ! `detail is User`는 저도 처음 보는 것 같아요 ㅎㅎ
참고할 블로그까지 찾아주시다니..! 정말 감사합니다 수현님 🙇🏻‍♀️ 블로그도 정독해보겠습니다 !!